omod-common/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/MainSubResourceController.java

```diff
@@ -165,4 +165,60 @@ public Object purge(@PathVariable("resource") String resource, @PathVariable("pa
 		return RestUtil.noContent(response);
 	}
 	
+	/**
+	 * @param parentUuid
+	 * @param reason
+	 * @param request
+	 * @throws Exception
+	 */
+	@RequestMapping(value = "/{resource}/{parentUuid}/{subResource}", method = RequestMethod.DELETE, params = "!purge")
+	@ResponseBody
+	public Object delete(@PathVariable("resource") String resource, @PathVariable("parentUuid") String parentUuid,
+	        @PathVariable("subResource") String subResource,
+	        @RequestParam(value = "reason", defaultValue = "web service call") String reason, HttpServletRequest request,
+	        HttpServletResponse response) throws ResponseException {

```

Comment:
the original handled delete /patient/{p_uuid}/allergy/{a_uuid} to delete a single allergy for patient
we needed delete /patient/{p_uuid}/allergy to delete all allergies for the patient